src/streaming_data_json.py

```diff
@@ -0,0 +1,123 @@
+import os
+import argparse
+import pandas as pd
+import json
+import logging
+from time import sleep
+from bson import json_util
+
+from kafka import KafkaProducer, KafkaAdminClient
+from kafka.admin import NewTopic
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-m",
+    "--mode",
+    default="setup",
+    choices=["setup", "teardown"],
+    help="Whether to setup or teardown a Kafka topic with driver stats events. Setup will teardown before beginning emitting events.",
+)
+parser.add_argument(
+    "-b",
+    "--bootstrap_servers",
+    default="localhost:9092",
+    help="Where the bootstrap server is",
+)
+
+args = parser.parse_args()
+
+NUM_DEVICES = 1
+
+
+def create_topic(admin, topic_name):
+    """
+        Create topic if not exists
+    """
+    try:
+        topic = NewTopic(name=topic_name, num_partitions=1, replication_factor=1)
+        admin.create_topic([topic])
+        logging.info(f"A new topic {topic_name} has been created!")
+    except Exception:
+        logging.info(f"Topic {topic_name} already exists. Skipping creation!")
+        pass
+
+
+def create_streams(servers):
+    """
+        Create streaming data to Kafka Topic
+    """
+
+    producer = None
+    admin = None
+    for _ in range (10): 
+        try:
+            producer = KafkaProducer(bootstrap_servers=servers)
+            admin = KafkaAdminClient(bootstrap_servers=servers)
+            break
+        except Exception as e:
+            logging.info(f"Trying to instantiate admin and producer with bosootstrap server {servers} with error {e}")
+            sleep(10)
+            pass
+
+    # send data
+    df = pd.read_parquet("data/2024/yellow_tripdata_2024-01.parquet")
```

Comment:
Don't hardcode your file here! Consider set it as a constant!

Reply:
For this file, I just want to simulate a data stream that is continuously being inserted into the database for processing.

Reply:
No exception!